students/eowyn/session08/circle.py

```diff
@@ -0,0 +1,131 @@
+"""
+nifty Circle class
+"""
+from math import pi
+
+class Circle:
+    def __init__(self, radius):
+        self.radius = radius
+
+    @property
+    def diameter(self):
+        return 2*self.radius
+
+    @diameter.setter
+    def diameter(self, val):
+        self.radius = val / 2
+
+    @property
+    def area(self):
+        return pi*self.radius**2
+
+    def __repr__(self):
+        return "Circle({})".format(self.radius)
+
+    def __str__(self):
+        return "Circle with radius: {}".format(self.radius)
+
+#  ---------------- NUMERIC OPERATORS --------------------------
+
+    def __add__(self, other):
+        # Add circles by adding their radii
+        newrad = self.radius + other.radius
+        return Circle(newrad)
+
+    def __mul__(self, factor):
+        newrad = self.radius * factor
+        return Circle(newrad)
+
+    def __rmul__(self, factor):
+        # reflective multiply... repeats code??
+        newrad = self.radius * factor
+        return Circle(newrad)
+
+    def __truediv__(self, factor):
+        newrad = self.radius / factor
+        return Circle(newrad)
+
+    def __rtruediv__(self, factor):
+        # reflective division... repeats code??
```

Comment:
it's reveresed -- is it giving the right answer? I woulndn't expect so!

but if it is repeated, you can re-use the previous definition:

def __rtrudiv__ = __truediv__

pretty cool, eh?